### PR TITLE
Socint 167 aims jwt

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Adds additional environment variables to allow for AIMS JWT creation and different urls for external (typeahead) and internal access of AIMS from RHUI

# What has changed
Removal of ADDRESS_INDEX_SVC_JWT envar
Addition of ADDRESS_INDEX_SVC_EXTERNAL_URL and ADDRESS_INDEX_SVC_KEY envars
New function to generate a JWT from the provided secret key

Has been tested against RHCuc in dev

New config and secret values have been added to all 3 environments, and terraform has been updated to pick them up

# Links
https://collaborate2.ons.gov.uk/jira/browse/SOCINT-167

### CODE REVIEW

**Review by rpcatling — CHANGES_REQUESTED** [review]
Couple of comments. In addition, "request a new access code" seems to fail locally for me, meaning RHcuc doesn't run locally.

**Review by SteveScorfield — APPROVED** [review]
Looks fine

File: app/config.py [hunk 1]
```diff
@@ -156,8 +158,9 @@ class TestingConfig:
     SESSION_AGE = ''
 
     ADDRESS_INDEX_SVC_URL = 'http://localhost:9000'
-    ADDRESS_INDEX_SVC_JWT = 'secret'
     ADDRESS_INDEX_EPOCH = ''
+    ADDRESS_INDEX_SVC_EXTERNAL_URL = 'http://localhost:9000'
+    ADDRESS_INDEX_SVC_KEY = 'c2VjcmV0PT0='  # Value must be base64 encoded, and multiple of 4 characters long unencoded
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply app/config.py]: its just dummy, apparently

File: app/security.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ async def on_prepare(request: web.BaseRequest, response: web.StreamResponse):
             value = '; '.join([
                 f"{section} {' '.join(content)} 'nonce-{request.csp_nonce}'"
                 if section in ADD_NONCE_SECTIONS else
-                f"{section} {' '.join(content)} {request.app['ADDRESS_INDEX_SVC_URL']}"
+                f"{section} {' '.join(content)} {request.app['ADDRESS_INDEX_SVC_EXTERNAL_URL']}"
```
[the remark this answers is not in the document]
Comment (rpcatling) [reply app/security.py:102]: confirmed

**Review by rpcatling — APPROVED** [review]

**Pull request merged**